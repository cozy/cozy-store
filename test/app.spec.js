'use strict'

/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'

import {
  App,
  mapStateToProps,
  mapDispatchToProps
} from '../src/ducks/components/App'

describe('App component only', () => {
  it('should be mounted correctly', () => {
    const initApp = jest.fn()
    const component = shallow(<App initApp={initApp} />).getElement()
    expect(component).toMatchSnapshot()
    expect(initApp.mock.calls.length).toBe(1)
  })

  it('should export mapStateToProps and mapDispatchToProps to allow customization', () => {
    expect(typeof mapStateToProps).toBe('function')
    expect(typeof mapDispatchToProps).toBe('function')
  })
})
