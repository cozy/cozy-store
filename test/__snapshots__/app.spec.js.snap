// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component only should be mounted correctly 1`] = `
<BreakpointsProvider>
  <Layout
    monoColumn={false}
  >
    <Alerter
      into="body"
    />
    <withI18n(withBreakpoints(withRouter(Sidebar))) />
    <Main>
      <Switch>
        <Route
          component={[Function]}
          path="/redirect"
        />
        <Route
          component={[Function]}
          path="/discover"
        />
        <Route
          component={[Function]}
          path="/myapps"
        />
        <React.Fragment>
          <Redirect
            exact={true}
            from="/"
            push={false}
            to="/discover"
          />
          <Redirect
            from="*"
            push={false}
            to="/discover"
          />
        </React.Fragment>
      </Switch>
    </Main>
    <IconSprite />
  </Layout>
</BreakpointsProvider>
`;
